src/app/detail/ui/Sidebar.tsx

```diff
@@ -184,10 +197,13 @@ export const Sidebar = ({
           </StyledText>
           <DatePickerComponent
             getDate={(date) => {
-              const dueDate = DateStringSchema.parse(formatDate(date))
-              updateTask({ dueDate })
+              const isoDate = formatDate(date as string)
+              setDueDate(date as string)
```

Comment:
Is there a reason why we are we removing a schema parse here and type casting instead?
The role of DateStringSchema is to make sure the date is in YYYY-MM-DD format. the `*date` functions could break if it isn't used.